extensions/multi-force/src/index.tsx

```diff
@@ -0,0 +1,155 @@
+      toast = await showToast({
+        style: Toast.Style.Animated,
+        title: "Refreshing your orgs.",
+      });
+    }
+
+    const flatOrgs = Array.from(orgs.values()).flat();
+
+    // Fetch the new list of orgs
+    const newOrgs = await getOrgList();
+
+    // Create a map from the new orgs for quick lookup
+    const newOrgsMap = new Map(newOrgs.map((org) => [org.username, org]));
+
+    // Filter the existing orgs to include only those that exist in the new list
+    const updatedOrgs = flatOrgs.filter((org) => newOrgsMap.has(org.username));
+
+    // Merge the existing org fields with the new org fields
+    const mergedOrgs = updatedOrgs.map((org) => ({
+      ...org,
+      ...newOrgsMap.get(org.username),
+    }));
+
+    // Add new orgs that are not present in the existing list
+    const additionalOrgs = newOrgs.filter(
+      (org) => !flatOrgs.some((existingOrg) => existingOrg.username === org.username),
+    );
+
+    // Combine merged orgs and additional new orgs
+    const combinedOrgs = [...mergedOrgs, ...additionalOrgs];
+
+    // Sort the combined list by alias
+    combinedOrgs.sort((a, b) => a.alias.localeCompare(b.alias));
+
+    // Update the orgs state
+    setOrgs(groupOrgs(combinedOrgs));
+    saveOrgs(combinedOrgs);
+    setIsLoading(false);
+    if (shouldShowToast) {
+      toast!.hide();
+    }
+  };
+
+  return Array.from(orgs.keys()).length === 0 && !isLoading ? (
+    <EmptyOrgList callback={refreshOrgs} />
+  ) : (
+    <List isLoading={isLoading}>
+      {Array.from(orgs.keys())
+        .sort()
+        .map((key, keyIndex) =>
+          orgs.get(key) && orgs.get(key)!.length > 0 ? (
+            <List.Section title={key} key={keyIndex}>
+              {orgs.get(key)!.map((org, index) => (
+                <List.Item
+                  key={index}
+                  icon={{ source: "Salesforce.com_logo.svg.png", tintColor: org.color ?? "#0000FF" }}
+                  title={org.label ? `${org.label} (${org.alias})` : org.alias}
+                  actions={
+                    <ActionPanel>
+                      <Action title="Open" onAction={() => handleOrgSelection(org.alias)} />
+                      <Action.Push
+                        title="Open Details"
+                        target={<DeveloperOrgDetails org={org} callback={refreshOrgs} />}
+                        shortcut={{ modifiers: ["cmd"], key: "return" }}
+                      />
+                      <Action
+                        title="Refresh Org List"
+                        onAction={() => refreshOrgs()}
+                        shortcut={{ modifiers: ["cmd"], key: "r" }}
+                      />
+                      <Action.Push
+                        title="Authenticate a New Org"
+                        target={<AuthenticateNewOrg callback={refreshOrgs} />}
+                        shortcut={{ modifiers: ["cmd"], key: "n" }}
+                      />
+                      <Action.Push
```

Comment:
Maybe it's better to use confirmAlert inside `<Action` instead of pushing to a new view, this way it's also possible to use `style={Action.Style.Destructive}`